src/backend/src/services/work-packages.services.ts

```diff
@@ -767,4 +767,76 @@ export default class WorkPackagesService {
 
     return workPackageTemplateTransformer(updatedWorkPackageTemplate);
   }
+
+  /**
+   * Deletes the Work Package template
+   * @param submitter The user who deleted the work package
+   * @param workPackageTemplateId The id of the work package template to be deleted
+   * @param organizationId The organization id that the user is in
+   */
+  static async deleteWorkPackageTemplate(
+    submitter: User,
+    workPackageTemplateId: string,
+    organizationId: string
+  ): Promise<void> {
+    // Verify submitter is allowed to delete work packages
+    if (!(await userHasPermission(submitter.userId, organizationId, isAdmin)))
+      throw new AccessDeniedAdminOnlyException('delete work package template');
+
+    const workPackageTemplate = await prisma.work_Package_Template.findUnique({
+      where: {
+        workPackageTemplateId
+      },
+      include: {
+        blocking: true
+      }
+    });
+
+    if (!workPackageTemplate) {
+      throw new NotFoundException('Work Package Template', workPackageTemplateId);
+    }
+
+    if (workPackageTemplate.dateDeleted) {
+      throw new DeletedException('Work Package Template', workPackageTemplateId);
+    }
+
+    if (workPackageTemplate.organizationId !== organizationId) {
+      throw new InvalidOrganizationException('Work Package Template');
+    }
+
+    const dateDeleted = new Date();
+
+    if (workPackageTemplate.blocking.length > 0) {
+      for (const template of workPackageTemplate.blocking) {
```

Comment:
This isnt quite good enough, we need to recursively delete the blockings of this work packages blockigs. Look at updateBlockings or getAllBlockingWorkPackages helper functions. You should make a similar one called deleteBlockings